### ISSUE TITLE
Enable user provided configuration

### ISSUE THREAD

**Issue by fabiocav** [Azure/azure-webjobs-sdk-script#1853]
This applies to 2.x (Core).

The runtime should pick up a configuration file in the function app root when building the configuration used by the application if one is present.

**Comment by MisinformedDNA** [comment]
Yes, I would like this to be as similar to ASP.NET Core as possible. A common approach helps everyone.

In the meantime, users can set up their own config using the code here: https://github.com/Azure/azure-functions-host/issues/2437#issuecomment-409082773
